bootstrap/018_warehouse_updates.sql

```diff
@@ -63,5 +64,29 @@ BEGIN
         RAISE;
 
     END;
+
+    SYSTEM$LOG_INFO('Starting warehouse size table refresh');
+    BEGIN
+        BEGIN TRANSACTION;
+            truncate INTERNAL.WAREHOUSE_SIZE_MAPPING;
+            SHOW WAREHOUSES;
+            insert into internal.warehouse_size_mapping select "name", "size" from TABLE(RESULT_SCAN(LAST_QUERY_ID()));
+        COMMIT;
+    END;
+    SYSTEM$LOG_INFO('Finished warehouse size table refresh');
     CALL INTERNAL.SET_CONFIG('WAREHOUSE_EVENTS_MAINTENANCE', CURRENT_TIMESTAMP()::string);
 END;
+
+CREATE OR REPLACE FUNCTION TOOLS.APPROX_CREDITS_USED(warehouse_name varchar, start_time timestamp, end_time timestamp)
+RETURNS NUMBER
+AS
+$$
+    TOOLS.WAREHOUSE_CREDITS_PER_MILLI((select warehouse_size from internal.warehouse_size_mapping t where t.warehouse_name = warehouse_name))*timestampdiff(millisecond, start_time, end_time)
```

Comment:
I see that `CREDITS_USED_CLOUD_SERVICES` is included in the query_history table function. Not sure how difficult it is to use a bunch of cloud services credits compared to warehouse credits. Intentional to omit `CREDITS_USED_CLOUD_SERVICES` here?